### ISSUE TITLE
DS4 Bluetooth Connectivity Issues

### ISSUE THREAD

**Issue by Bovrinox** [Ryochan7/DS4Windows#1459]
Hi there, I am having terrible trouble connecting my PS4 controller to my PC. I have installed DS4Windows, and the drivers as per instructions. I have paired my controller following the steps on the driver install page of DS4Windows. However, although it shows in the bluetooth options page as Wireless Controller: Paired, Whenever I press the power button on the controller, it just flashes a white light 6 times and goes off again. It shows up in device manager as Wireless Controller under the Bluetooth tab. 
When i try the debug version of ds4, I get an error message, which i have attached as a log file.
[DS4Windows Debug Error.txt](https://github.com/Ryochan7/DS4Windows/files/5113812/DS4Windows.Debug.Error.txt)

It works perfectly if I connect with a usb cable but unfortunately this is impractical due to the distance i am seated from my machine, and the fact that long enough usb cables dont seem to last 5 minutes. Any help resolving this issue would be greatly appreciated!
Thanks 
Matt

**Desktop (please complete the following information):**
 - OS: Windows 10 Home 1909
 - DS4Windows Version 2.1.5 x86

**Additional context**

**Comment by Bovrinox** [comment]
I have checked these things, the only one that wasnt the latest version was the Visaulc ++ 
Unfortunately this has made no difference what so ever. The controller still refuses to do anything.  In fact i cant even get either of my 2 official sony controllers to even pair now. 
Could it be that my usb bluetooth thing is too old? i think it is only bluetooth 4.
I am currently waiting for a proper bluetooth card so i can free up some usb ports, ill see if i have any more luck then and report back
Thanks a lot for trying to help

**Comment by mika-n** [comment]
Hmm... Do you have 64bit Windows10 OS? Are you sure you don't have mixed 32bit and 64bit DS4Windows installation files?
Try re-downloading the 64bit version of DS4Windows and unzip it to a new empty folder (or cleanup everyfile in the old installation folder).
https://github.com/Ryochan7/DS4Windows/releases

Some PC systems may complain that the file was downloaded from an unknown web site and using FileExplorer zip tool to unzip the file EXE/DLL files may be in "blocked" state. Take a look at properties of DS4Windows.exe and Nefarious.Vigem.Client.dll file (right mouse btn and properties in FileExplorer) and make sure "blocked" option is not set.

Try unzipping the app to somewhere else than below c:\program files (or x86 version of the folder). Use for example c:\apps\ds4windows\    Some PCs have extra permission issues in programFiles folder and it an app is not installed using a installation manager then files copied below programFiles folder may be in blocked state/limited permissions.

**Comment by Bovrinox** [comment]
Well Ive got it partially working now.
As you correctly pointed out, I have 64 bit win10 and had downloaded the 32 bit version, like a muppet. However reinstalling it with the correct version didnt immediately fix the issue. I moved the usb bluetooth adapter to the usb port on the front of my case, and now it works. Both are usb 3 so not sure why that made any difference!
However, it doesn't work fully. It wont register when navigating menus in game for some reason, although it operates as it should while actually playing. the buttons work, x to select, o to go back etc but neither the d pad nor sticks will move the cursor. No idea why this is, I have tried playing with various settings but to no avail. I am having a lot of problems trying to get it to work with anything on steam too. Either with or without ds4windows running.
Thanks for trying to help though

**Comment by mika-n** [comment]
Double controller issue? See the following page about "Hide DS4 Controller" option. Or if it doesn't work in your PC then HidGuardia driver is an alternative solution to hide the original physical controller from games and leaving only the virtual gamepad visible (this avoids the double-input issue in some games).
https://github.com/Ryochan7/DS4Windows/wiki/Exclusive-Mode-(Hide-DS4-Controller-config-option)-tips-and-issues

If sticks don't seem to do anything then check deadzone value in LS/RS axis. If the deadzone is accidentally 1.00 (or more) then the stick doesn't do anything (the whole axis range is one big deadzone). ControllerReadings tab page should reveal this (red circle is the deadzone region).

https://github.com/Ryochan7/DS4Windows/wiki/Troubleshooting

**Comment by Bovrinox** [comment]
Ah toggling the hide ds4 controller seems to have done the trick actually.  Only reason i hadn't tried that was because on my old pc I had bother with that, but it was doing exactly what you'd expect, moving 2 menu items per 1 click haha.
I have also just installed a proper bluetooth/wifi card into my pc rather than using usb dongles, so that may of helped. It does connect a lot faster now.
I did in fact have to increase the dead zone for both sticks up to 0.10 to stop them creeping slightly.
Thank you so much for your help getting this sorted, and also for your patience in dealing with me!
Thanks
Bov
